test/constructorio-synonyms-one-way.js

```diff
@@ -0,0 +1,587 @@
+      }
+      return resolve(response);
+    });
+  });
+}
+
+// Helper method - remove one way synonym for specified parent phrase
+function removeTestOneWaySynonym(phrase) {
+  const constructorio = new Constructorio(testConfig);
+
+  return new Promise((resolve, reject) => {
+    constructorio.removeOneWaySynonym({
+      phrase,
+    }, (err, response) => {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(response);
+    });
+  });
+}
+
+describe.only('ConstructorIO - One Way Synonyms', () => {
+  // Introduce latency to avoid throttling issues
+  beforeEach((done) => {
+    setTimeout(() => {
+      done();
+    }, 200);
+  });
+
+  describe('addOneWaySynonym', () => {
+    const oneWaySynonymPhrase = 'cheese';
+
+    after((done) => {
+      // Clean up test one way synonyms
+      removeTestOneWaySynonym(oneWaySynonymPhrase).then(done).catch(done);
+    });
+
+    it('should add a one way synonym', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.addOneWaySynonym({
+        phrase: oneWaySynonymPhrase,
+        child_phrases: [
+          { phrase: 'parmesan' },
+          { phrase: 'gouda' },
+          { phrase: 'mozzarella' },
+        ],
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+
+    it('should return an error when supplying a phrase that already exists', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.addOneWaySynonym({
+        phrase: oneWaySynonymPhrase,
+        child_phrases: [
+          { phrase: 'parmesan' },
+          { phrase: 'gouda' },
+          { phrase: 'mozzarella' },
+        ],
+      }, (err, response) => {
+        expect(err).to.be.an('object');
+        expect(err).to.have.property('message', 'A one way synonym with the parent phrase "cheese" already exists.');
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+
+    it('should return an error when supplying child phrases of incorrect type', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.addOneWaySynonym({
+        phrase: oneWaySynonymPhrase,
+        child_phrases: 'pizza',
+      }, (err, response) => {
+        expect(err).to.be.an('object');
+        expect(err).to.have.property('message', 'child_phrases must be a list');
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+
+    it('should return an error when not supplying a child phrases property', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.addOneWaySynonym({}, (err, response) => {
+        expect(err).to.be.an('object');
+        expect(err).to.have.property('message', 'child_phrases is a required field of type list');
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
```

Comment:
Maybe a test for when a `phrase` isn't supplied?

Reply:
Added!